src/strands/tools/registry.py

```diff
@@ -279,6 +279,34 @@ def register_tool(self, tool: AgentTool) -> None:
                 list(self.dynamic_tools.keys()),
             )
 
+    def replace(self, tool_name: str, new_tool: AgentTool) -> None:
```

Comment:
I think we don't need to take in the `tool_name` here given that `new_tool` name has to match the old one, correct?

Or alternatively phrased - why do we think we need the `tool_name` passed in?